src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -431,16 +617,44 @@ zocl_create_cu(struct drm_zocl_dev *zdev)
 
 	return 0;
 err:
-	kfree(cu_info);
-	zocl_destroy_cu(zdev);
+	zocl_destroy_cu_slot(zdev, slot->slot_idx);
 	return err;
 }
 
 static inline bool
-zocl_xclbin_same_uuid(struct drm_zocl_dev *zdev, xuid_t *uuid)
+zocl_xclbin_same_uuid(struct drm_zocl_slot *slot, xuid_t *uuid)
+{
+	return (zocl_xclbin_get_uuid(slot) != NULL &&
+	    uuid_equal(uuid, zocl_xclbin_get_uuid(slot)));
+}
+
+/*
+ * Return the slot pointer for the given xclbin uuid as an Input.
+ *
+ * @param       zdev:	zocl device structure
+ * @param       id:	xclbin uuid
+ *
+ * @return      domian pointer on success, NULL on failure.
+ */
+struct drm_zocl_slot *
+zocl_get_slot(struct drm_zocl_dev *zdev, uuid_t *id)
```

Comment:
@saifuddin-xilinx  What if same xclbin with same uuid is loaded onto multiple slot? this routine will always return first slot with matching uuid. Is this scenario supported?

Reply:
This scenario is only valid for AIE cases. But this PR is targeted for only multi -slot PL cases only. 
As we verified things on zocl and also single xclbin cases in xocl.
We will cover AIE and built-in PS Kernel in next PR with xocl changes